docs/deployment.md

```diff
@@ -50,21 +50,71 @@ Bootstrap cdk, replacing `<profile name>`
 npx cdk bootstrap --profile <profile name>
 ```
 
-## Deploy the cdk stack
-
-Deploy the cdk stack, replacing `<profile name>` and env `<production or staging>`
+Then, run VPC configuration that cannot be done via cdk
 
 ```sh
-# Run VPC configuration that cannot be done via cdk
 npm run configureVpc
+```
+
+## Deploy the cdk stack
+
+The cdk stack deploys the docker container on ECS (Elastic Container Service), within a VPC (Virtual
+Private Network) that is provisioned by OIT and is required for ECS deployment. Within the VPC, an
+ALB (application load balancer) serves as an entrypoint for communications from outside the VPC to
+the services running within. OIT requires communicating with this internal ALB over HTTPS. We need
+to manually generate and upload the SSL certificate for this internal ALB to communicate over HTTPS.
 
-npm run cdk:deploy --profile <profile name> --context env=<production or staging>
-# Add `--require-approval never` to skip the yes/no
+Thus, deploying this stack is weirdly cyclical:
+
+1.  The first-ever deployment should be a http deployment. This generates the random
+    cloudformation-generated internal ALB DNS, e.g.
+    internal-doula-assistant-alb-1234567890.us-east-1.elb.amazonaws.com, where `1234567890` is a
+    random number
+2.  Then, `scripts/generate_self_signed_cert.py` should be run to generate a self-signed SSL
+    certificate using the random DNS generated by cloudformation. The certificate should then be
+    uploaded to to AWS Certificate Manager
+3.  Only after the internal ALB is generated, and the certificate created and uploaded, can we then
+    run the HTTPS version of this stack that uses the certificate
+
+It would be great if there's a better way to do this, but this author is currently not aware of one.
+
+### First-time HTTP deployment
+
+First, deploy the cdk stack with a http internal ALB, replacing `<profile name>` and env
+`<production or staging>`
+
+```sh
+npm run cdk:deploy -- --profile <profile name> --context env=<production or staging>
 ```
 
 The output `DoulaAssistantStack.LoadBalancerUrl` should be accessible from an
 [AWS CloudShell environment created within the VPC](https://docs.aws.amazon.com/cloudshell/latest/userguide/using-cshell-in-vpc.html).
 
+### Generate and upload the SSL certificate for the internal ALB
+
+Using the output `DoulaAssistantStack.LoadBalancerUrl`, create a self-signed SSL certificate,
+replacing `<load balancer domain>` and `<email>`. The load balancer domain is the SAN and not the
+common name because it is likely too long to fit the 64-character limit on common names. The SSL
+certificate can be self-signed because it's just used for TLS communication.
```

Comment:
Maybe specify, just for TLS between internal and external load balancers. So this won't be the cert that user browsers see, just the one that the external load balancer sees